woocommerce/woocommerce
# Feature/32158 complete task list card with feedback

### All Submissions:






### Changes proposed in this Pull Request:

This PR end up being relatively large, mostly because the changes spread across multiple area's, here are the main changes:
- Add a new `CustomerFeedbackSimple` component to the `customer-effort-score` package, I also made sure the FeedbackModal was exported so we can make use of it.
     - I also converted some of it to TS and had to add the `types` to the package.json so build-types was used.
- Created a new `CompletedHeaderWithCES` that I used in both the new task lists (SectionedTaskList, and TaskList within the two column tasks folder)
- I added a `showCESFeedback` prop to the task list definition that checks if tracking is enabled and if the score has already been submitted.
- This also required changes in the Onboarding data package, so the redux store gets updated correctly without refreshing the page.

Closes #32158

![finished-task-list-header](https://user-images.githubusercontent.com/2240960/165108277-bbed0ab6-dcfe-4b5d-b29a-2888697ec091.gif)

### How to test the changes in this Pull Request:

Use the latest version of [WooCommerce Admin Test Helper](https://github.com/woocommerce/woocommerce-admin-test-helper) plugin to toggle features, toggle `tasklist-setup-experiment-1` to true.

1. Finish the onboarding, enabling tracking, and don't pick any of the free features 
2. Finish all the steps in the task list
3. Once you completed the last task the `You've completed store setup` header should show with the CES `How was your experience?`.
4. If you click `Difficult` or `Somewhat Difficult` emoji's (should show that description as a tooltip on hover), a modal should show with the emoji pre-selected. It should also trigger a `wcadmin_ces_view` track, can be seen in your browser console having tracks output enabled - `localStorage.setItem( 'debug', 'wc-admin:*' );`
5. Click cancel or 'x', it should do nothing and the `How was your experience? should remain displayed`
6. Click one of the other 3 emoji's, it should show, `We appreciate your feedback!` for a second or so and then disappear. It should also have submitted the score through tracks by emitting a `wcadmin_ces_feedback` event with the action as `store_setup` with the correct score.
7. Refresh the page, the CES part should now show up anymore.
8. If you want to try the above again with different score you can delete the `woocommerce_ces_shown_for_actions` option to have it show up again.
9. Now click the Ellipsis and try both options, Show task list should show the task list, and Hide this should hide it (both should persist on refresh)

### Other information:




### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### FOR PR REVIEWER ONLY:
